File: src/main/java/ua/softgroup/matrix/server/supervisor/producer/json/UserTimeResponse.java
package ua.softgroup.matrix.server.supervisor.producer.json;

/**
 * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
 */
public class UserTimeResponse extends TimeResponse {

    private long userId;

    public UserTimeResponse() {
    }

    public UserTimeResponse(long userId, int workSeconds, int idleSeconds, double idlePercentage) {
        this.userId = userId;
        this.totalWorkTimeSeconds = workSeconds;
        this.totalIdleTimeSeconds = idleSeconds;
        this.totalIdlePercentage = idlePercentage;
    }

    public long getUserId() {
        return userId;
    }

    public void setUserId(long userId) {
        this.userId = userId;
    }

}

File: src/main/scala/ua/softgroup/matrix/server/supervisor/producer/json/UserTimeResponse.scala
package ua.softgroup.matrix.server.supervisor.producer.json

import scala.beans.BeanProperty

/**
  * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
  */
class UserTimeResponse extends TimeResponse {

  @BeanProperty
  var userId: Long = _

  def this(userId: Long, workSeconds: Int, idleSeconds: Int, idlePercentage: Double) {
    this()
    this.userId = userId
    this.totalWorkTimeSeconds = workSeconds
    this.totalIdleTimeSeconds = idleSeconds
    this.totalIdlePercentage = idlePercentage
  }

  override def toString: String = {
    "UserTimeResponse(userId=%d, totalWorkTimeSeconds=%d, totalIdleTimeSeconds=%d, totalIdlePercentage=%f)"
      .format(userId, totalWorkTimeSeconds, totalIdleTimeSeconds, totalIdlePercentage)
  }

}
